src/proxy/protocol.rs

```diff
@@ -35,11 +37,49 @@ impl Protocol {
             // We didn't want to keep parsing headers, just validate that
             // the first line is HTTP1.
             Ok(_) | Err(httparse::Error::TooManyHeaders) => {
+                println!("^^^^^ is H1");
                 return Some(Protocol::Http1);
             }
             _ => {}
         }
 
+        // Detect if this is a Kafka traffic
+
+        // Header:
+        //     Request/response Size => INT32
+        //     api_key => INT16
+        //     api_version => INT16
+        //     correlation_id => INT32
+        //     client_id => NULLABLE_STRING (first two bytes as the length of the client_id string)
+        const KAFKA_REQUEST_HEADER_LENGTH: usize = 4 + 2 + 2 + 4 + 2;
+        println!("length {}", bytes.len());
+        if bytes.len() >= KAFKA_REQUEST_HEADER_LENGTH {
+            let request_size =      BigEndian::read_i32(&bytes);
+            let api_key =           BigEndian::read_i16(&bytes[4..]);
+            let api_version =       BigEndian::read_i16(&bytes[6..]);
+            let correlation_id =    BigEndian::read_i32(&bytes[8..]);
+            let len_client_id =     BigEndian::read_i16(&bytes[12..]);
```

Comment:
Just so you're aware, CI builds on this repository are checked using [rustfmt](https://github.com/rust-lang/rustfmt), a Rust code formatting tool. Branches which are not formatted according to rustfmt's rules will fail on CI. 

It may make your life easier to download `rustfmt` and run it on changes branches prior to pushing them. Personally, I have VS Code configured to run `rustfmt` every time I save a file.